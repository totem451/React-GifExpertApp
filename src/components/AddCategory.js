import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategorias}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {

        console.log(e.target.value);
        setInputValue(e.target.value);

    };

    const handleSubmit = (e) => {

        e.preventDefault();

        console.log('Submit hecho');

        if (inputValue.trim().length > 0) {

            // setCategorias(c => [...c, inputValue]);
            setCategorias(c => [inputValue, ...c]);
            setInputValue(' ');

        }

    };

    return (

        <>

            <h1>{inputValue}</h1>
            <h2>Add Category</h2>
            <form onSubmit = {handleSubmit}>
                <input 
                    type = 'text'
                    value = {inputValue}
                    onChange = {handleInputChange}
                />
            </form>

        </>

    )
}

AddCategory.propTypes = {
    setCategorias: PropTypes.func.isRequired
}