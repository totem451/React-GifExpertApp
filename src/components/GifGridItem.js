import React from 'react'

export const GifGridItem = ({id, title, url}/*props*/) => {

    console.log(id, title, url/*props*/)


    return (
        <div className = 'card animate__animated animate__fadeInDown'>
            <img src = {url} alt = {title} />
            <p>{title}</p>
        </div>
    )

}
