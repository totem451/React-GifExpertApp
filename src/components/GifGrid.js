import React/*, { useState, useEffect }*/ from 'react';
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs';
// import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ categoria }) => {

    // const [count, setCount] = useState(0)

    // const [images, setImages] = useState([]);

    // const state = useFetchGifs();
    const {data:images, loading} = useFetchGifs( categoria );
    // console.log(data);
    // console.log(loading);

    // useEffect(() => {
    //     getGifs(categoria)
    //         .then(setImages);
    // }, [categoria])

    // const getGifs = async() => {

    //     const url = `https://api.giphy.com/v1/gifs/search?q=${encodeURI(categoria)}&limit=10&api_key=XQYohDnNz7C9bpUK8df50bU8ybj42vmY`;
    //     const resp = await fetch(url);
    //     // resp.ok
    //     const {data} = await resp.json();

    //     // console.log(data);

    //     const gifs = data.map(img => {

    //         return {
    //             id: img.id,
    //             title: img.title,
    //             url: img.images?.downsized_medium.url
    //         }

    //     })

    //     console.log(gifs);
    //     setImages(gifs);

    // };

    return (  

        // <>
        //     <h3>{categoria}</h3>
        //     <div className='card-grid'>
        //         {/* <h3>{count}</h3>
        //         <button onClick = {() => setCount(count + 1)}></button> */}
        //         {/* <ol>
        //             {
        //                 images.map(img => (
        //                     <li key = {img.id}>{img.title}</li>
        //                 ))
        //             }
        //             {
        //                 images.map(({id, title}) => (
        //                     <li key = {id}>{title}</li>
        //                 ))
        //             }
        //         </ol> */}
        //         {
        //             images.map(img => (
        //                 <GifGridItem
        //                     key={img.id}
        //                     img={img}
        //                     {...img}
        //                 />
        //             ))
        //         }
        //     </div>
        // </>

        <>
            <h3 className = 'animate__animated animate__fadeInDown'>{categoria}</h3>

            {/* {loading ? 'Cargando...' : 'Data cargada'} */}
            { loading && <p className = 'animate__animated animate__flash'>Loading...</p> }
            
            <div className = 'card-grid'>

                {
                    images.map(img => (
                        <GifGridItem
                            key={img.id}
                            img={img}
                            {...img}
                        />
                    ))
                }
            </div>

        </>
    )
}

GifGrid.propTypes = {
    categoria: PropTypes.string.isRequired
}