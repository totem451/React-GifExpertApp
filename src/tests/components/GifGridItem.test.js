import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';
import '@testing-library/jest-dom';
import '@wojtekmaj/enzyme-adapter-react-17'

describe('Pruebas en <GifGridItem /> ', () => {

    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg';
    
    const wrapper = shallow( <GifGridItem  title = { title } url = { url }/> );

    test('debe mostrar el componente correctamente ', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test('debe tener un parrafo con el titulo ', () => {
        
        const p = wrapper.find('p')

        expect( p.text().trim() ).toBe( title )

    });
    
    test('debe tener la imagen igual al url y alt de los props ', () => {
        
        const img = wrapper.find('img'); // una imagen
        // const img = wrapper.find('#img'); // mas de una imagen, id
        // const img = wrapper.find('.img'); // mas de una imagen, clase
        // const img = wrapper.find('img').at; // mas de una imagen, varias
        console.log( img.html() );
        console.log( img.props().src );
        console.log( img.prop('src'));

        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );

    });

    test('debe tener animate_-fadeInDown ', () => {
        
        const div = wrapper.find('div');

        expect( div.prop('className').includes('animate__fadeInDown') ).toBe( true );

    });
    
})
