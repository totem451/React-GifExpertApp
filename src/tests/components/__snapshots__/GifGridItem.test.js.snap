// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem />  debe mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
